### PR TITLE
API+Indexer refactor

### PR DESCRIPTION
**Pull request by alexsporn**
- Refactors the Indexer API to use an opaque cursor
- Removed `data`-envelope from API v2
- Re-structured info-endpoint and added new protocol parameters to the response

### CODE REVIEW

File: plugins/restapi/v2/types.go [hunk 1]
```diff
@@ -37,6 +32,28 @@ type infoResponse struct {
...
+	ReferencedRate float64 `json:"referencedRate"`
+}
+
+// infoResponse defines the response of a GET info REST API call.
+type infoResponse struct {
+	// The name of the node software.
+	Name string `json:"name"`
+	// The semver version of the node software.
+	Version string     `json:"version"`
+	Status  nodeStatus `json:"status"`
```
Comment (muXxer) [inline plugins/restapi/v2/types.go]:
```suggestion
	// The current status of this node.
	Status  nodeStatus `json:"status"`
```

**Review by muXxer — APPROVED** [review]

File: plugins/indexer/plugin.go [hunk 2]
```diff
@@ -102,31 +107,31 @@ func configure() {
...
 	}
 
-	configureEvents()
+	onLedgerUpdatedWorkerPool = workerpool.New(func(task workerpool.Task) {
+		updateIndexer(task.Param(0).(milestone.Index), task.Param(1).(utxo.Outputs), task.Param(2).(utxo.Spents))
+		task.Return(nil)
+	}, workerpool.WorkerCount(workerCount), workerpool.QueueSize(workerQueueSize), workerpool.FlushTasksAtShutdown(true))
+
+	onLedgerUpdated = events.NewClosure(func(index milestone.Index, newOutputs utxo.Outputs, newSpents utxo.Spents) {
+		onLedgerUpdatedWorkerPool.TrySubmit(index, newOutputs, newSpents)
```
Comment (muXxer) [inline plugins/indexer/plugin.go]: `TrySubmit` would mean we can loose ledger updates :scream:

**Review by muXxer — CHANGES_REQUESTED** [review]

File: plugins/indexer/plugin.go (hunk 2 above)
Comment (alexsporn) [reply plugins/indexer/plugin.go]: `workerQueueSize = 10000` not really...

Comment (muXxer) [reply plugins/indexer/plugin.go]: If the queue is full, the node should wait, not drop the event!

Comment (alexsporn) [reply plugins/indexer/plugin.go]: Ok, I'll change it to `Submit`

**Review by muXxer — APPROVED** [review]

**Pull request merged**